### PR TITLE
Adds RX11 stimulant delivery system.

### PR DESCRIPTION
**Pull request by Rivet-the-Zombie**
Must be worn before use.

Requires stimulant modules to function.

Spawns in the same places and at the same frequency as the RX12 healing
injector.

Immediately restores stamina to maximum while removing the 'winded'
effect, adds weak painkiller and strong stimulant effects while reducing
fatigue, and imparts a special effect which temporarily boosts stats and
improves stamina regeneration at the cost of a chemical comedown.

Renames the RX12's boost effect in order to better differentiate it from
the new RX11 stimulant effect.

Also found a bug: the effect name displayed in the speed display of the
@ screen is always the name of the  comedown effect for whatever is
boosting your speed. I did not fix this bug to avoid cluttering this PR.

### CODE REVIEW

**Comment by Zireael07** [comment]
Cool!

**Comment by Coolthulhu** [comment]
Stamina regen should probably be a generic drug effect, so that meth and other stimulant drugs can do it instead of giving superhuman speed.

**Comment by Rivet-the-Zombie** [comment]
>Stamina regen should probably be a generic drug effect, so that meth and other stimulant drugs can do it instead of giving superhuman speed.

Really, meth and such ought to be removing fatigue and boosting the heck out of morale but that's about it; RL chemical stimulants don't do jack for your stamina or make you move any faster.

File: src/game.cpp [hunk 1]
```diff
@@ -1141,10 +1141,16 @@ bool game::do_turn()
...
-        // But mouth encumberance interferes.
+        // But mouth encumbrance interferes.
         u.stamina += std::max( 1, 10 - (u.encumb(bp_mouth) / 10) );
         // TODO: recovering stamina causes hunger/thirst/fatigue.
     }
+    
+    // 3d4 bonus stamina from active stimpack stamina-boost.
+    if( u.stamina < u.get_stamina_max() && u.has_effect("stimpack") && 
+        u.get_effect_dur("stimpack") > 50 ) {
+        u.stamina += std::max( 3, 12 );
```
Comment (Coolthulhu) [inline src/game.cpp]: Looks like you wanted a `rng` here.

Also, that's a lot. You can run forever with it.

And it has no side effects, except the very minor comedown.

Comment (kevingranade) [inline src/game.cpp]: As we're starting to add synthetic performance-enhancing drugs, I'd love to
start removing the nonsense buffs from recreational drugs.

**Comment by Rivet-the-Zombie** [comment]
Stamina boost during effect is now 2d4 instead of 3d4.

**Comment by Coolthulhu** [comment]
Merged, but it will need adjustments later on.
The instant stamina recovery will need some cost, for example adding a lot of fatigue during comedown (like with meth). And preferably a large health cost.

**Pull request merged**